zeppelin-user-impersonation[3016].md

```diff
@@ -0,0 +1,152 @@
+
+How to set up user impersonation at Zeppelin Shell Interpreter level with shiro authentication can found from [here](https://zeppelin.apache.org/docs/0.7.0/manual/userimpersonation.html)
+What the different kinds of Interpreter binding modes is well discussed in the upcoming version 0.8.0 of Zeppelin [documentation](https://zeppelin.apache.org/docs/0.8.0-SNAPSHOT/usage/interpreter/interpreter_binding_mode.html)
+
+Before setting up the Zeppelin User Impersonation make sure you have integrated your zeppelin and your cluster with AD/LDAP, below is the shiro.ini for AD authentication: </br>
+<i>
+[users] </br>
+###List of users with their password allowed to access Zeppelin </br>
+###To use a different strategy (LDAP / Database / ...) check the shiro doc at http://shiro.apache.org/configuration.html#Configuration-INISections </br>
+admin = admin, admin  </br>
+user1 = user1, role1, role2 </br>
+user2 = user2, role3 </br>
+user3 = user3, role2 </br>
+###Sample LDAP configuration, for user Authentication, currently tested for single Realm </br>
+[main] </br>
+###A sample for configuring Active Directory Realm </br>
+activeDirectoryRealm = org.apache.zeppelin.realm.ActiveDirectoryGroupRealm </br>
+activeDirectoryRealm.systemUsername = CN=Administrator,CN=Users,DC=ADSERVER,DC=com </br>
+activeDirectoryRealm.systemPassword = loginpassword </br>
+activeDirectoryRealm.searchBase = CN=Users,DC=ADSERVER,DC=com </br>
+activeDirectoryRealm.url = ldap://ldap.ad.ADSERVER.com:389 </br>
+activeDirectoryRealm.authorizationCachingEnabled = true </br>
+sessionManager = org.apache.shiro.web.session.mgt.DefaultWebSessionManager </br>
+securityManager.sessionManager = $sessionManager </br>
+
+###If caching of user is required then uncomment below lines </br>
+cacheManager = org.apache.shiro.cache.MemoryConstrainedCacheManager </br>
+securityManager.cacheManager = $cacheManager </br>
+###securityManager.sessionManager = $sessionManager </br>
+###86,400,000 milliseconds = 24 hour </br>
+securityManager.sessionManager.globalSessionTimeout = 86400000 </br>
+shiro.loginUrl = /api/login </br>
+[roles] </br>
+role1 = * </br>
+role2 = * </br>
+role3 = * </br>
+admin = * </br>
+[urls] </br>
+###This section is used for url-based security. </br>
+###You can secure interpreter, configuration and credential information by urls. Comment or uncomment the below urls that you want to hide. </br>
+###anon means the access is anonymous. </br>
+###authc means Form based Auth Security </br>
+###To enfore security, comment the line below and uncomment the next one </br>
+/api/version = anon </br>
+#/api/interpreter/** = authc, roles[admin] </br>
+#/api/configurations/** = authc, roles[admin] </br>
+#/api/credential/** = authc, roles[admin] </br>
+#/** = anon </br>
+/** = authc </br>
+</i>
+
+<b>Below are the configurations required from the environment:</b> </br>
+•	On the Zeppelin server node make sure to add the below to <i>/etc/sudoers</i>: </br>
```

Comment:
Is it true ? I am using mac, so don't need this. Not sure whether linux need that.

Reply:
i have tested this with Linux which is needed, but not sure about Mac.